server/deps/repository_adapters/occurrence_repository_adapter.py

```diff
@@ -2,78 +2,75 @@
+    ClassroomRepositoryDep,
+)
+from server.deps.repository_adapters.schedule_repository_adapter import (
+    ScheduleRepositoryDep,
+)
 from server.deps.session_dep import SessionDep
-from server.models.database.class_db_model import Class
 from server.models.database.schedule_db_model import Schedule
-from server.repositories.class_repository import ClassRepository
-from server.repositories.classroom_repository import ClassroomRepository
+from server.models.http.requests.allocate_request_models import AllocateSchedule
 from server.repositories.occurrence_repository import OccurrenceRepository
-from server.repositories.schedule_repository import ScheduleRepository
+from server.services.security.buildings_permission_checker import (
+    building_permission_checker,
+)
+from server.utils.must_be_int import must_be_int
 
 
 class OccurrenceRepositoryAdapter:
-    def __init__(self, building: BuildingDep, session: SessionDep):
-        self.building = building
+    def __init__(
+        self,
+        session: SessionDep,
+        user: UserDep,
+        classroom_repo: ClassroomRepositoryDep,
+        schedule_repo: ScheduleRepositoryDep,
+    ):
         self.session = session
+        self.user = user
+        self.classroom_repo = classroom_repo
+        self.schedule_repo = schedule_repo
 
     def allocate_schedule(self, schedule_id: int, classroom_id: int) -> Schedule:
-        schedule = ScheduleRepository.get_by_id_on_building(
-            schedule_id=schedule_id,
-            building=self.building,
-            session=self.session,
-        )
-        classroom = ClassroomRepository.get_by_id_on_building(
-            building=self.building, id=classroom_id, session=self.session
-        )
+        schedule = self.schedule_repo.get_by_id(schedule_id)
+        classroom = self.classroom_repo.get_by_id(classroom_id)
         OccurrenceRepository.allocate_schedule(
             schedule=schedule, classroom=classroom, session=self.session
         )
         self.session.commit()
         self.session.refresh(schedule)
         return schedule
 
-    def allocate_class(self, class_id: int, classroom_id: int) -> Class:
-        class_ = ClassRepository.get_by_id_on_building(
-            id=class_id,
-            building=self.building,
-            session=self.session,
-        )
-        classroom = ClassroomRepository.get_by_id_on_building(
-            building=self.building, id=classroom_id, session=self.session
-        )
-        OccurrenceRepository.allocate_class(
-            class_=class_, classroom=classroom, session=self.session
-        )
+    def allocate_schedule_many(
+        self, schedule_classroom_pairs: list[AllocateSchedule]
+    ) -> None:
+        for pair in schedule_classroom_pairs:
+            schedule = self.schedule_repo.get_by_id(pair.schedule_id)
+            building_permission_checker(self.user, schedule.class_.subject.buildings)
+
+            if pair.classroom_id == -1:
```

Comment:
Não gosto da ideia de usar -1 para desalocar, eu deixaria opcional esse campo e caso seja None ele desaloca. Ainda não vi como você fez no front, mas normalmente isso funcionaria bem com o front.

Reply:
é que no front eu to usando o valor None como "não fazer nada". Ele só fica -1 se o usuário explicitamente marcar que quer desalocar